### ISSUE TITLE
dada2 handling low quality reverse read

### ISSUE THREAD

**Issue by Kennlu0402** [benjjneb/dada2#556]
Hi there, 

I am using dada2 to deal with reads had bad quality (especially on the reverse reads) as following:

![image](https://user-images.githubusercontent.com/42954326/45008662-7ad06380-b036-11e8-8ee0-51cd953d07b8.png)

so I kept the truncLen for the reverse as 250 (and 275 for the forwards) to keep enough overlay between forwards and reverse reads. 

At the end, dada2 was able to generate a feature table. But when I blast some sequences, it shows only 92% similarity to the sequences in NCBI. I am wondering whether dada2 can handle the reverse reads with that much low quality reads. Since using Trimmomatic, it gives me only several 100s reads from both surviving by filtering with Q20. 

Or what about reverse reads with low quality like the following from another set of data? I am running dada2 on this dataset too. 
![image](https://user-images.githubusercontent.com/42954326/45008805-2ed1ee80-b037-11e8-8a39-5562b9df4bc6.png)

Thank you in advance.

**Comment by Kennlu0402** [comment]
Hi Benjjneb, 

Sure. The primer set is 27F/534R targeting the V1-3 region of 16S rRNA, so the expected length would be around 510bp, and the sequencing are Miseq 2x300bp. The primers (and adpators) were trimmed by cutadapt prior to dada2.  
As there were high amount of noise (the quality dropped dramatically, I had some samples were even worse) than expected, I am not sure whether too much bad quality reads can affect the accuracy of error correction? 

Thank you.

**Comment by benjjneb** [comment]
> As there were high amount of noise (the quality dropped dramatically, I had some samples were even worse) than expected, I am not sure whether too much bad quality reads can affect the accuracy of error correction?

Yes, it can. The error-correction algorithm is robust to lower quality, up to a point. But when you have too much low quality sequence that you can't trim off, the ability to distinguish true sequences from errors starts to degrade (technically this depends on fraction of error-free reads in the data).

The amplicon you've sequenced is very long (~550 nts including the sequenced primers) and the read quality is not good. This is a very challenging situation for achieving accurate single-nucleotide resolution. An alternative would be to work with the forward reads alone.

**Comment by Kennlu0402** [comment]
Thanks Benjjneb.  That is exactly what I was wondering. 

If I blast the result sequences generated from dada2, in one analysis (let's call it set A), it gives me around 97-99% similarity to NCBI database, but 91% similarity to the database in another set (set B). My understanding is set A gives me the true error-free reads while set B may be affected by the low-quality reads. Does that sound right to you?

**Comment by benjjneb** [comment]
> If I blast the result sequences generated from dada2, in one analysis (let's call it set A), it gives me around 97-99% similarity to NCBI database, but 91% similarity to the database in another set (set B). My understanding is set A gives me the true error-free reads while set B may be affected by the low-quality reads. Does that sound right to you?

While that is one possible interpretation, it may also be that set B has more sequences that are not (well) represented in our reference databases. Are both datasets from the same type of environment?

**Comment by Kennlu0402** [comment]
They are from experiments with different set up.  Sure that is definitely one possibility too, I guess if most mismatch are located in the middle of the sequences, I would expect it is due to the low quality reads exceed the limit of error-correction. 

Thank you

**Comment by larssnip** [comment]
We have a problem that I suspect may be along the same lines as mentioned above (long amplicon+lots of sequencing errors).

We observe that after de-noising R1 and R2 reads, there are plenty of high-abundance sequence types in each group, but when merging them most of the high-abundance sequence types (from both R1 and R2) are not merged! This raises a question:

We followed the steps of the turorial, which means error-correction of R1 and R2 reads separately, and then merging. Why not merge reads first, and then do the error-model training and correction? I can see some arguments for "correcting" R1 and R2 reads first, because if this is (highly) successful, the merging of "error-free" reads will be easier. However, NOT using the pairwise information inherent in the fastq-files is also a waste of information.

**Comment by benjjneb** [comment]
> Why not merge reads first, and then do the error-model training and correction?

[From the paper](https://doi.org/10.1038/nmeth.3869): "One such difference is that the DADA2 pipeline performs merging of paired-end reads after denoising. This is because the core denoising algorithm uses the empirical relationship between the quality score and the error rates. When reads are merged, this relationship will differ between the forward-only, overlapping, and reverse-only portions of the merged read. That variation interferes with the denoising algorithm, and therefore greater accuracy can be achieved by denoising before merging, albeit at some computational cost."
